### PR TITLE
Fix forward accounting

### PR DESCRIPTION
**Pull request by rustyrussell**
By not actually testing a real migration, I missed that it would have issues.  This fixes the migration.

Those who've already upgraded (to master) have lost their forward data for closed channels :(

Changelog-None: Fixes issues since last release

### CODE REVIEW

File: doc/lightning-listforwards.7.md [hunk 1]
```diff
@@ -25,12 +25,12 @@ RETURN VALUE
...
 
 - **in\_channel** (short\_channel\_id): the channel that received the HTLC
-- **in\_htlc\_id** (u64): the unique HTLC id the sender gave this
 - **in\_msat** (msat): the value of the incoming HTLC
 - **status** (string): still ongoing, completed, failed locally, or failed after forwarding (one of "offered", "settled", "local_failed", "failed")
 - **received\_time** (number): the UNIX timestamp when this was received
+- **in\_htlc\_id** (u64, optional): the unique HTLC id the sender gave this (not present if incoming channel was closed before ugprade to v22.11)
 - **out\_channel** (short\_channel\_id, optional): the channel that the HTLC (trying to) forward to
-- **out\_htlc\_id** (u64, optional): the unique HTLC id we gave this when sending
+- **out\_htlc\_id** (u64, optional): the unique HTLC id we gave this when sending (may be missing even if out_channel is present, for old forwards before v22.11)
```
Comment (cdecker) [inline doc/lightning-listforwards.7.md:33]: So we're switching version numbering scheme then, ok :-)

**Review by cdecker — APPROVED** [review]
ACK 8f8516826947f0104fc4108513057cc0ca559622

**Comment by cdecker** [comment]
Test needs an exemption for elements where the DB snapshot doesn't match the expected network.

**Comment by rustyrussell** [comment]
This Time For Sure!

Ack 0c06844e4

File: doc/lightning-listforwards.7.md (hunk 1 above)
Comment (rustyrussell) [reply doc/lightning-listforwards.7.md:33]: I guessed!

**Pull request merged**